internal/controllers/synthesis/exec_io.go

```diff
@@ -11,9 +11,44 @@ import (
 	krmv1 "github.com/Azure/eno/pkg/krm/functions/api/v1"
 	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+func buildPodInput(comp *apiv1.Composition, syn *apiv1.Synthesizer) ([]byte, error) {
+	bindings := map[string]*apiv1.Binding{}
+	for _, b := range comp.Spec.Bindings {
+		b := b
+		bindings[b.Key] = &b
+	}
+	refs := map[string]*apiv1.Ref{}
+	for _, r := range syn.Spec.Refs {
+		r := r
+		refs[r.Key] = &r
+	}
+
+	inputs := []*unstructured.Unstructured{}
+	for key, r := range refs {
+		b, ok := bindings[key]
+		if !ok {
+			return nil, fmt.Errorf("input %q is referenced, but not bound", key)
+		}
+		input := apiv1.NewInput(key, apiv1.InputResource{
+			Name:      b.Resource.Name,
+			Namespace: b.Resource.Namespace,
```

Comment:
I think we should default the namespace to the composition's namespace. Since typically resource references are contained to the referencing resource's namespace e.g. secrets/configmaps from pods, it seems natural to allow omitting the namespace.

I like that we can reference across namespaces though. It's all too common to get stuck replicating secrets between namespaces just to reference them from pods

Reply:
Discussed offline, we don't want to default because it can cause problems with non-namespaced resources.